cli/tests/test_branch_command.rs

```diff
@@ -1348,6 +1348,80 @@ fn test_branch_list_tracked() {
     "###);
 }
 
+#[test]
+fn test_branch_list_conflicted() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    // Set up remote
+    let git_repo_path = test_env.env_root().join("git-repo");
+    let git_repo = git2::Repository::init(git_repo_path).unwrap();
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["git", "remote", "add", "origin", "../git-repo"],
+    );
+    let create_remote_commit = |message: &str, data: &[u8], ref_names: &[&str]| {
+        let signature =
+            git2::Signature::new("Some One", "some.one@example.com", &git2::Time::new(0, 0))
+                .unwrap();
+        let mut tree_builder = git_repo.treebuilder(None).unwrap();
+        let file_oid = git_repo.blob(data).unwrap();
+        tree_builder
+            .insert("file", file_oid, git2::FileMode::Blob.into())
+            .unwrap();
+        let tree_oid = tree_builder.write().unwrap();
+        let tree = git_repo.find_tree(tree_oid).unwrap();
+        // Create commit and branches in the remote
+        let git_commit_oid = git_repo
+            .commit(None, &signature, &signature, message, &tree, &[])
+            .unwrap();
+        for name in ref_names {
+            git_repo.reference(name, git_commit_oid, true, "").unwrap();
+        }
+    };
+
+    // Fetch new commit without auto tracking. No local branches should be
+    // created.
+    create_remote_commit(
+        "commit 1",
+        b"content 1",
+        &[
+            "refs/heads/main",
+            "refs/heads/feature1",
+            "refs/heads/feature2",
+        ],
+    );
+    test_env.add_config("git.auto-local-branch = false");
```

Comment:
nit: perhaps, these setup codes are copied from somewhere, and can be simplified?

If we don't care about the remote branches, the easiest way to create conflicts is:
```
jj new 'root()' -m a
jj new 'root()' -m b
jj branch create foo -rREV_A
jj branch create foo -rREV_B --at-op=@-
```

Reply:
Yes, they were partially copied from another branch test. The smaller setup looks good, I'll change it to that.